src/map/src/items/inventory.cpp

```diff
@@ -114,7 +127,7 @@ ReturnValue Items::add_item(EntitySystem& entitySystem, RoseCommon::Entity entit
         }
     }
     RoseCommon::Packet::SrvSetItem::IndexAndItem index; index.set_index(pos);
-    index.set_item(entitySystem.item_to_item<RoseCommon::Packet::SrvSetItem>(item));
+    index.set_item(entitySystem.item_to_item<RoseCommon::Packet::SrvSetItem>(inv.items[pos]));
```

Comment:
Not sure why that was changed, inv.items[pos] == item

Reply:
When add_item gets a stackable item, we're adding the count to the current item of the same type & id which is inv.items[pos] (and not item which is new).
When the server used 'item', the stacking was bugged :)

Reply:
oh right! Then we need to delete the `item` entity after we add it, otherwise we'll have dangling entities lying around in the storage, something like `entitySystem.delete_entity(item);` should do the trick